src/app/app.component.ts

```diff
@@ -1,13 +1,96 @@
-import { Component } from '@angular/core';
-import { RouterOutlet } from '@angular/router';
+import { Component, OnInit } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { Student } from './student';
+import { Course } from './course';
+import { StudentListComponent } from "./student-list/student-list.component";
+import { StudentDetailComponent } from "./student-detail/student-detail.component";
 
 @Component({
-  selector: 'app-root',
-  standalone: true,
-  imports: [RouterOutlet],
-  templateUrl: './app.component.html',
-  styleUrl: './app.component.css'
+    selector: 'app-root',
+    standalone: true,
+    templateUrl: './app.component.html',
+    styleUrl: './app.component.css',
+    imports: [
+        CommonModule,
+        StudentListComponent,
+        StudentDetailComponent
+    ]
 })
-export class AppComponent {
-  title = 'jdf-30x30-step2';
+export class AppComponent implements OnInit {
+  students: Student[] = [];
+
+  selectedStudent: Student = this.getStudentById(-1);
+
+  Course = Course;
+
+  ngOnInit(): void {
+    this.students = [
+      {
+        id: 0,
+        firstName: "Joe",
+        lastName: "Smith",
+        isActive: true,
+        courses: [
+          Course.MachineLearning
+        ]
+      },
+      {
+        id: 1,
+        firstName: "Jane",
+        lastName: "Doe",
+        isActive: false,
+        courses: [
+          Course.IntroToHtml,
+          Course.FunctionalJavaScript
+        ]
+      }
+    ];
+  }
+
+  selectStudent(student: Student): void {
+    this.selectedStudent = student;
+  }
+
+  newStudent(): void {
+    this.selectedStudent = this.getStudentById(-1);
+  }
+
+  deleteStudent(student: Student): void {
```

Comment:
This works form removing, but you can also do this, since you're just removing an element from an array.  Either works. Figured I would mention to get you familiar with other array functions.

this.students = this.students.filter((s) =>s.id !== student.id);